File: delete-nodes-from-linked-list-present-in-array/delete-nodes-from-linked-list-present-in-array.java
class Solution {
    public ListNode modifiedList(int[] nums, ListNode head) {
        Set<Integer> valuesToRemove = new HashSet<>();
        for(int num : nums) valuesToRemove.add(num);
        while(head != null && valuesToRemove.contains(head.val)) head = head.next;
        if(head == null) return null;
        ListNode curr = head;
        while(curr.next != null){
            if(valuesToRemove.contains(curr.next.val)){
                curr.next = curr.next.next;
            }
            else{
                curr = curr.next;
            }
        }
        return head;
    }
}

File: delete-nodes-from-linked-list-present-in-array/delete-nodes-from-linked-list-present-in-array.cpp

class Solution {
public:
    ListNode* modifiedList(vector<int>& nums, ListNode* head) {
        unordered_set<int> valuesToRemove(nums.begin(), nums.end());
        while(head != NULL && valuesToRemove.count(head->val) > 0){
            ListNode *temp = head;
            head = head->next;
            delete temp;
        }
        if(head == NULL) return NULL;
        ListNode *curr = head;
        while(curr->next != NULL){
            if(valuesToRemove.contains(curr->next->val)){
                ListNode *temp = curr->next;
                curr->next = curr->next->next;
                delete temp;
            }
            else{
                curr = curr->next;
            }
        }
        return head;
    }
};